main.js

```diff
@@ -163,16 +173,69 @@ function executeCommand(cmd) {
     }
     if ((cmd == 'E' || cmd == 'e') && playerLocation.col < 9) { 
         playerLocation.col = playerLocation.col + 1;
-    } 
+    }
     if ((cmd == 'U' || cmd == 'u') && playerLocation.floor < 9) { 
-        playerLocation.floor = playerLocation.floor + 1; 
+        playerLocation.floor = playerLocation.floor + 1;
     }
     if ((cmd == 'D' || cmd == 'd') && playerLocation.floor > 0) { 
         playerLocation.floor = playerLocation.floor - 1;
     }
+    if (cmd == 'A' || cmd == 'a') {
+        var room = map[playerLocation.floor][playerLocation.row][playerLocation.col];
+        if (room.monsterStats) {
+            if (playerLocation.agility > room.monsterStats.agility) {
+                fight(playerLocation, room.monsterStats);
+                fight(room.monsterStats, playerLocation);
+            } else {
+                fight(room.monsterStats, playerLocation);
+                fight(playerLocation, room.monsterStats);
+            }
+
+            if (room.monsterStats.health <= 0) {
+                display("You have vanquished the " + room.key);
+                awardXP();
+                delete room.monsterStats;
+            }
+            if (playerLocation.health <= 0) {
+                display('You are dead!');
+            }
+        } else {
+            display("What are you swinging at?")
+        }
+    }
     drawMap();
 }
 
+function fight(a, d) {
+    if (a.health <= 0) {
+        return;
+    }
+    display(a.key + ' attacks!')
+    var dodgeRate = Math.max((d.agility - a.agility) / (a.agility + d.agility), .1 );
+    var attack = Math.max(a.attack - d.defense, 1);
+    if (Math.random() < dodgeRate) {
+        display(d.key + ' dodged the attack');
+    } else {
+        d.health -= attack;
+    }
+}
+
+function awardXP() {
+    playerLocation.xp += 1;
+    if (playerLocation.xp % 5 == 0) {
+        playerLocation.health = 100;
+        playerLocation.attack += getRandomInt(20);
+        playerLocation.defense += getRandomInt(20);
+        playerLocation.agility += getRandomInt(20);
+    }
+}
+
+function display(message) {
+    var outputDiv = document.getElementById("output");
+    outputDiv.innerHTML = outputDiv.innerHTML + '<br>' + message;
+    outputDiv.scrollTop = outputDiv.scrollHeight;
+}
+
```

Comment:
makes da playder and moonsters attacks dieffer dapendoing on da tyope and da amor.
change the code so different monsters such as master oogway have better health and attack but worse speed.